src/service.js

```diff
@@ -368,6 +404,79 @@ function getServices(holder) {
   return services;
 }
 
+
+/**
+ * Whether the specified service implements `Disposable` interface.
+ * @param {!Object} service
+ * @return {boolean}
+ */
+export function isDisposable(service) {
+  return service.dispose !== undefined;
+}
+
+
+/**
+ * Asserts that the specified service implements `Disposable` interface and
+ * typecasts the instance to `Disposable`.
+ * @param {!Object} service
+ * @return {!Disposable}
+ */
+export function assertDisposable(service) {
```

Comment:
I'd expect it to be inlined after `dev().assert()` is removed. Right? Otherwise it doesn't really belong in `log` since the interface itself is defined here at this time.